p2p/simulations/adapters/inproc.go

```diff
@@ -372,3 +381,53 @@ func netPipe() (net.Conn, net.Conn, error) {
 	p1, p2 := net.Pipe()
 	return p1, p2, nil
 }
+
+func tcpPipe() (net.Conn, net.Conn, error) {
+	cl := make(chan net.Conn)
+	cd := make(chan net.Conn)
+	start := make(chan *net.TCPAddr)
+
+	go func(listener chan net.Conn, start chan *net.TCPAddr) {
+		found := false
+		for !found {
+			// assign random free port to current listener
```

Comment:
Listen on port `0` and the kernel will pick a random port for you.

Reply:
Thanks, I forgot I could do that! Fixed.